transports/tcp/src/lib.rs

```diff
@@ -368,7 +371,20 @@ where
         socket.bind(&socket_addr.into())?;
         socket.listen(self.config.backlog as _)?;
         socket.set_nonblocking(true)?;
-        TcpListenStream::<T>::new(id, socket.into(), self.port_reuse.clone())
+        let listener: TcpListener = socket.into();
+        let local_addr = listener.local_addr()?;
+        let if_watcher = if local_addr.ip().is_unspecified() {
+            Some(IfWatcher::new()?)
+        } else {
+            self.port_reuse.register(local_addr.ip(), local_addr.port());
+            let listen_addr = ip_to_multiaddr(local_addr.ip(), local_addr.port());
+            self.pending_events.push_back(TransportEvent::NewAddress {
+                listener_id: id,
+                listen_addr,
+            });
+            None
+        };
+        TcpListenStream::<T>::new(id, listener, if_watcher, self.port_reuse.clone())
```

Comment:
Perhaps two different constructors would be a good idea here?

Reply:
Like `TcpListenStream::new_single_interface` and `TcpListenStream::new_all_interfaces`?
They only differ by one line, so not sure if it makes sense.
(The above logic for enqueuing a `NewAddress` event can not be moved into the constructor).
Or did I misunderstand you?

Reply:
It would enable an early return but other than that, it is equivalent yeah! Up to you :)

Reply:
Not sure if I understand what you mean. @thomaseizinger wdyt of 4a63d84?

Reply:
Looks good to me, thank you!